lib/Analysis/Allocation.cpp

```diff
@@ -115,8 +115,7 @@ SmallVector<unsigned> getScratchConfigForReduce(triton::ReduceOp op) {
 // TODO: extend beyond scalars
 SmallVector<unsigned> getScratchConfigForAtomicRMW(triton::AtomicRMWOp op) {
   SmallVector<unsigned> smemShape;
-  auto ptrTy = op.ptr().getType();
-  if (auto tensorType = ptrTy.dyn_cast<RankedTensorType>()) {
+  if (auto tensorType = op.ptr().getType().dyn_cast<RankedTensorType>()) {
```

Comment:
we could use isa<>() instead of dyn_cast when the returned value is not to be used.